src/plugins/mergeCells/mergeCells.js

```diff
@@ -279,6 +279,67 @@ class MergeCells extends BasePlugin {
     return auto ? true : this.validateSetting(newMergedCellInfo);
   }
 
+  /**
+   *
+   */
+  toggleMergeOnSelection() {
+    const currentRange = this.hot.getSelectedRangeLast();
+
+    if (!currentRange) {
+      return;
+    }
+
+    currentRange.setDirection('NW-SE');
+
+    const {from, to} = currentRange;
+
+    this.toggleMerge(currentRange);
+    this.hot.selectCell(from.row, from.col, to.row, to.col, false);
+  }
+
+  /**
+   * Merge the selection provided as a cell range.
+   *
+   * @param {CellRange} [cellRange] Selection cell range.
+   */
+  mergeSelection(cellRange) {
+    if (!cellRange) {
+      cellRange = this.hot.getSelectedRangeLast();
+
+      if (!cellRange) {
+        return;
+      }
+    }
+
+    cellRange.setDirection('NW-SE');
+
+    const {from, to} = cellRange;
+
+    this.unmergeRange(cellRange, true);
+    this.mergeRange(cellRange);
+    this.hot.selectCell(from.row, from.col, to.row, to.col, false);
+  }
+
+  /**
+   * Merge the selection provided as a cell range.
+   *
+   * @param {CellRange} [cellRange] Selection cell range.
+   */
+  unmergeSelection(cellRange) {
+    if (!cellRange) {
+      cellRange = this.hot.getSelectedRangeLast();
+
+      if (!cellRange) {
+        return;
+      }
+    }
+
```

Comment:
The `cellRange` checking can be simplified by adding the selection range as a default value, something like:
```js
mergeSelection(cellRange = this.hot.getSelectedRangeLast()) {
  if (!cellRange) {
    return;
  }
...
```